Import libraries

In [1]:
import requests
from bs4 import BeautifulSoup
import pandas as pd

Send request to the website <br>
And create Beautiful Soup object <br>

In [2]:
url = requests.get('https://www.eshop-apisehk.com/categories/foot-protection?locale=en')
soup = BeautifulSoup(url.content , 'html.parser') 

Get all products on the page

In [3]:
products = soup.select('div.Product-info')
products

[<div class="Product-info"> <div class="Product-title Label mix-primary-text" style="">Nitti 21980S Work Shoes</div> <div class="Label-price sl-price">HK$420.00</div> </div>,
 <div class="Product-info"> <div class="Product-title Label mix-primary-text" style="">PUMA FUSE KNIT BLACK WNS LOW Safety Shoes</div> <div class="Label-price sl-price">HK$1,250.00</div> </div>,
 <div class="Product-info"> <div class="Product-title Label mix-primary-text" style="">PUMA FRONTSIDE LOW Composite Toecap Composite Midsole Safety Shoes</div> <div class="Label-price sl-price">HK$1,050.00</div> </div>,
 <div class="Product-info"> <div class="Product-title Label mix-primary-text" style="">PUMA AIRTWIST BLUE LOW Composite Toecap Composite Midsole Safety Shoes</div> <div class="Label-price sl-price">HK$1,100.00</div> </div>,
 <div class="Product-info"> <div class="Product-title Label mix-primary-text" style="">PUMA AIRTWIST BLACK LOW Composite Toecap Composite Midsole Safety Shoes</div> <div class="Label-pri

In [4]:
# length of 'products', i.e. number of products
len(products)

24

In [5]:
# first product
products[0]

<div class="Product-info"> <div class="Product-title Label mix-primary-text" style="">Nitti 21980S Work Shoes</div> <div class="Label-price sl-price">HK$420.00</div> </div>

Get product name

In [6]:
first_product_name = products[0].select('div.Product-title.Label.mix-primary-text')
print(first_product_name)

[<div class="Product-title Label mix-primary-text" style="">Nitti 21980S Work Shoes</div>]


In [7]:
first_product_name[0].text

'Nitti 21980S Work Shoes'

Get product price

In [8]:
first_product_price = products[0].select('div.Label-price.sl-price ')
first_product_price

[<div class="Label-price sl-price">HK$420.00</div>]

In [9]:
first_product_price = first_product_price[0].text
first_product_price

'HK$420.00'

In [10]:
first_product_price = first_product_price.replace('HK$', '')
first_product_price

'420.00'

In [11]:
first_product_price = float(first_product_price)
first_product_price

420.0

Get product link

In [16]:
links = soup.select('a.Product-item')
links

[<a class="Product-item" ga-product='{"id":"627f315ce3b3600018b08f3e","sku":"21980S","variations":"[{\"key\":\"6281ed8cd98b7656461941b9\",\"sku\":\"21980S\",\"fields_translations\":{\"en\":[\"\"],\"zh-hant\":[\"UK2 / EU35\"]}}]","title":"Nitti 21980S Work Shoes"}' href="https://www.eshop-apisehk.com/products/nitti-21980s-work-shoes" ng-click='sendGaProductClick("627f315ce3b3600018b08f3e", "21980S", ["{\"key\":\"6281ed8cd98b7656461941b9\",\"sku\":\"21980S\",\"fields_translations\":{\"en\":[\"\"],\"zh-hant\":[\"UK2 / EU35\"]&amp;#125;&amp;#125;"], "Nitti 21980S Work Shoes")' product-id="627f315ce3b3600018b08f3e"> <div class="Image-boxify Product-image js-boxify-image"> <div class="Image-boxify-image js-image-boxify-image sl-lazy-image" style="background-image:url(https://shoplineimg.com/5ea1384c68412700240163df/627f02499f487a0018a1977e/800x.jpg?)"></div> <div class="Image-boxify-image js-image-boxify-image sl-lazy-image second-image" style="background-image:url(https://shoplineimg.com/5e

In [17]:
links[0]['href']

'https://www.eshop-apisehk.com/products/nitti-21980s-work-shoes'

Print product name and price

In [19]:
for product in products:
    product_name = product.select('div.Product-title.Label.mix-primary-text')[0].text
    product_price = float(product.select('div.Label-price.sl-price')[0].text.replace('HK$', ''))
    print(product_name, product_price)

Nitti 21980S Work Shoes 420.0


ValueError: could not convert string to float: '1,250.00'

We get an error message <br>
ValueError: could not convert string to float: '1,250.00' <br>
The string '1,250' cannot be converted to float because of the comma (thousand separator) <br>
We have to replace it with a blank to fix this <br>

In [ ]:
for product in products:
    product_name = product.select('div.Product-title.Label.mix-primary-text')[0].text
    product_price = float(product.select('div.Label-price.sl-price')[0].text.replace('HK$', '').replace(',', ''))
    print(product_name, product_price)

Print the product links

In [20]:
for link in links:
    print(link['href'])

https://www.eshop-apisehk.com/products/nitti-21980s-work-shoes
https://www.eshop-apisehk.com/products/puma-fuse-knit-black-wns-low-safety-shoes
https://www.eshop-apisehk.com/products/puma-frontside-low-composite-toecap-composite-midsole-safety-shoes
https://www.eshop-apisehk.com/products/627ccf06d5ccb30027196659
https://www.eshop-apisehk.com/products/puma-airtwist-black-low-composite-toecap-composite-midsole-safety-shoes
https://www.eshop-apisehk.com/products/puma-sierra-nevada-mid
https://www.eshop-apisehk.com/products/puma-charge-orange-low
https://www.eshop-apisehk.com/products/puma-conquest-stone-high-composite-toecap-and-composite
https://www.eshop-apisehk.com/products/smaat-sfm160-safety-shoes-with-steel-toecap-and-steel-midsole
https://www.eshop-apisehk.com/products/smaat-sfc200-safety-shoes-with-composite-cap-and-anti-penetration-midsole
https://www.eshop-apisehk.com/products/smaat-sfc220-safety-shoes-with-composite-cap-and-anti-penetration-midsole
https://www.eshop-apisehk.com

<h4> Store the data in a Pandas DataFrame </h4> <br>
We can create empty lists <br>
Then append each item to the list <br>
And finally create a Pandas DataFrame using a dictionary <br>

In [22]:
product_name_list = []
product_price_list = []
product_links_list = []
for product in products:
    product_name_list.append(product.select('div.Product-title.Label.mix-primary-text')[0].text)
    product_price_list.append(float(product.select('div.Label-price.sl-price')[0].text.replace('HK$', '').replace(',', '')))
for link in links: 
    product_links_list.append(link['href'])
df = pd.DataFrame({'name': product_name_list, 'price': product_price_list, 'link': product_links_list})

In [23]:
df

name   price  \
0                             Nitti 21980S Work Shoes   420.0   
1           PUMA FUSE KNIT BLACK WNS LOW Safety Shoes  1250.0   
2   PUMA FRONTSIDE LOW Composite Toecap Composite ...  1050.0   
3   PUMA AIRTWIST BLUE LOW Composite Toecap Compos...  1100.0   
4   PUMA AIRTWIST BLACK LOW Composite Toecap Compo...  1100.0   
5   PUMA SIERRA NEVADA MID Composite Toecap and Co...  1690.0   
6   PUMA CHARGE ORANGE LOW Composite Toecap and Co...  1100.0   
7   PUMA CONQUEST STONE HIGH Composite Toecap and ...  1300.0   
8   SMAAT SFM160 Safety Shoes with Steel Toecap an...   350.0   
9   SMAAT SFC200 Safety Shoes with Composite Cap a...   455.0   
10  SMAAT SFC220 Safety Shoes with Composite Cap a...   440.0   
11  Albatros ANTELAO QL LOW Composite toe cap and ...  1030.0   
12                        OTTER OWT993KW Safety Shoes  1130.0   
13                        OTTER OWT900KW Safety Shoes  1065.0   
14                      OTTER OWT805KW-R Safety Shoes  1700.0   
15              King's KL331Z Anti-slip Working Shoes   328.0   
16             King's KWS807Z Anti-slip Working Shoes   352.0   
17  King's KWD807 Steel Toe Steel Midsole Safety S...   480.0   
18  King's KWD806 Steel Toe Steel Midsole Safety B...   495.0   
19              King's KJ404Z Anti-slip Working Shoes   328.0   
20  King's KWD805C Steel Toe Steel Midsole Safety ...   592.0   
21  King's KWD805 Steel Toe Steel Midsole Safety B...   592.0   
22  King's KWS901 Steel Toe Steel Midsole Safety S...   392.0   
23  King's KWS803 Steel Toe Steel Midsole Safety S...   488.0   

                                                 link  
0   https://www.eshop-apisehk.com/products/nitti-2...  
1   https://www.eshop-apisehk.com/products/puma-fu...  
2   https://www.eshop-apisehk.com/products/puma-fr...  
3   https://www.eshop-apisehk.com/products/627ccf0...  
4   https://www.eshop-apisehk.com/products/puma-ai...  
5   https://www.eshop-apisehk.com/products/puma-si...  
6   https://www.eshop-apisehk.com/products/puma-ch...  
7   https://www.eshop-apisehk.com/products/puma-co...  
8   https://www.eshop-apisehk.com/products/smaat-s...  
9   https://www.eshop-apisehk.com/products/smaat-s...  
10  https://www.eshop-apisehk.com/products/smaat-s...  
11  https://www.eshop-apisehk.com/products/albatro...  
12  https://www.eshop-apisehk.com/products/otter-o...  
13  https://www.eshop-apisehk.com/products/otter-o...  
14  https://www.eshop-apisehk.com/products/otter-o...  
15  https://www.eshop-apisehk.com/products/kings-k...  
16  https://www.eshop-apisehk.com/products/kings-k...  
17  https://www.eshop-apisehk.com/products/kings-k...  
18  https://www.eshop-apisehk.com/products/kings-k...  
19  https://www.eshop-apisehk.com/products/kings-k...  
20  https://www.eshop-apisehk.com/products/kings-k...  
21  https://www.eshop-apisehk.com/products/kings-k...  
22  https://www.eshop-apisehk.com/products/kings-k...  
23  https://www.eshop-apisehk.com/products/kings-k...

In [24]:
df.to_excel('Safety_Shoes.xlsx', index=False)